src/dune_rules/super_context.ml

```diff
@@ -258,8 +258,12 @@ let extend_action t ~dir build =
 let make_rule t ?mode ?loc ~dir { Action_builder.With_targets.build; targets } =
   let build = extend_action t build ~dir in
   Rule.make
+    ?loc
     ?mode
-    ~info:(Rule.Info.of_loc_opt loc)
+    ~info:
+      (match loc with
+       | None -> Internal
+       | Some _ -> From_dune_file)
```

Comment:
You wrote this as `if Loc.is_none loc then Internal else From_dune_file` and slightly differently here. Should we factor out this piece of code?

Reply:
Mostly because `loc` is option here while it's not where I used `Loc.is_none`. I could give the optional value a default here and unify though. E.g.

```
let make_rule ?(loc=Loc.none) .. =
  let info = if Loc.is_none then Internal else From_dune_file
```

I'm not sure how to exactly factor out this piece of code. I guess one option would be to have `Rule.make` infer the `info` if it's not provided from the location?

Reply:
Ah, I didn't notice they were slightly different. Let's keep things as is then.